src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,61 @@
+package duke.command;
+
+import duke.TaskList;
+import duke.Ui;
+import duke.exception.InvalidTaskNumberException;
+import duke.task.Task;
+
+/**
+ * A class for the delete command.
+ */
+public class DeleteCommand extends Command {
+
+    /** Index of the task to be deleted in string form. */
+    private final String index;
+
+    /**
+     * Constructor for the DeleteCommand class.
+     *
+     * @param index Index of the task to be deleted in string form.
+     */
+    public DeleteCommand(String index) {
+        this.index = index;
+    }
+
+    /**
+     * Executes the delete command.
+     *
+     * @param taskList The taskList storing all tasks.
+     * @param ui       The ui responsible for interactions with the user.
+     * @throws InvalidTaskNumberException If the followup text after the command is not a valid integer.
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui) throws InvalidTaskNumberException {
+        if (index.equals("") || !isInteger(index) || (Integer.parseInt(index) - 1) < 0
+                || (Integer.parseInt(index) - 1) >= taskList.size()) {
```

Comment:
Perhaps you might want to implement a function to check whether the index is valid? 
Eg. `isValidIndex(taskList)` to avoid overcomplicating the expression highlighted.